community/rust/APKBUILD

```diff
@@ -81,22 +97,66 @@ _sharedir="usr/share/rust"
 ldpath="/$_rlibdir"
 
 export RUST_BACKTRACE=1
-export RUSTC_CRT_STATIC="false"
 # Convince libgit2-sys to use the distro libgit2.
 export LIBGIT2_SYS_USE_PKG_CONFIG=1
 
+# rust checksums files in vendor/, but we have to patch a few files...
+_clear_vendor_checksums() {
+	sed -i 's/\("files":{\)[^}]*/\1/' vendor/$1/.cargo-checksum.json
+}
+
 prepare() {
 	default_prepare
 
+	sed -i /LD_LIBRARY_PATH/d src/bootstrap/bootstrap.py
+
+	_clear_vendor_checksums libc
+	_clear_vendor_checksums lzma-sys
+
 	# Remove bundled dependencies.
 	rm -Rf src/llvm/
 }
 
 build() {
+	# We have to add new arches in multiple steps:
```

Comment:
You don't need multiple steps here. If you patch rust to include the `aarch64-alpine-linux-musl` target (which you've already done), and add `target = [..., "aarch64-alpine-linux-musl"]` and `host = [..., "aarch64-alpine-linux-musl"]` to your `config.toml`, rustc will cross-compile a native compiler with your desired triple, no intermediate "`unknown`" triple needed.

Reply:
This is the `config.toml` I use for compiling the Adélie binaries. All of the toolchains are built with musl-cross-make.
```toml
[llvm]
optimize = true
release-debuginfo = false
assertions = false
static-libstdcpp = true
ninja = true
targets = "AArch64;ARM;PowerPC;X86"
experimental-targets = ""

[build]
build = "powerpc64-foxkit-linux-musl" # Or you could build everything on x86_64, it makes no difference
host = [
  "aarch64-foxkit-linux-musl",
  "armv7-foxkit-linux-musleabihf",
  "i586-foxkit-linux-musl",
  "powerpc-foxkit-linux-musl",
  "powerpc64-foxkit-linux-musl",
  "x86_64-foxkit-linux-musl"
]
target = [
  "aarch64-foxkit-linux-musl",
  "armv7-foxkit-linux-musleabihf",
  "i586-foxkit-linux-musl",
  "powerpc-foxkit-linux-musl",
  "powerpc64-foxkit-linux-musl",
  "x86_64-foxkit-linux-musl"
]
cargo = "/usr/src/rust/rust-1.34.0/bin/cargo" # The previous binary...
rustc = "/usr/src/rust/rust-1.34.0/bin/rustc"
#cargo = "/usr/bin/cargo" # ...or the installed toolchain, it makes no difference
#rustc = "/usr/bin/rustc"
docs = true
compiler-docs = false
submodules = false
python = "/usr/bin/python3"
locked-deps = true
vendor = false
full-bootstrap = false
extended = true
tools = ["cargo", "clippy", "rls", "rustfmt", "src"]
verbose = 0
sanitizers = false
profiler = false
cargo-native-static = true

[rust]
optimize = true
codegen-units = 1
debug-assertions = false
debuginfo = false
debuginfo-lines = false
backtrace = true
default-linker = "gcc"
channel = "stable"
rpath = true # So the binaries work wherever they're extracted
ignore-git = true
jemalloc = false

[dist]
src-tarball = true

[target.aarch64-foxkit-linux-musl]
cc = "aarch64-linux-musl-gcc"
cxx = "aarch64-linux-musl-g++"
linker = "aarch64-linux-musl-gcc"

[target.armv7-foxkit-linux-musleabihf]
cc = "armv7-linux-musleabihf-gcc"
cxx = "armv7-linux-musleabihf-g++"
linker = "armv7-linux-musleabihf-gcc"

[target.i586-foxkit-linux-musl]
cc = "i586-linux-musl-gcc"
cxx = "i586-linux-musl-g++"
linker = "i586-linux-musl-gcc"

[target.powerpc-foxkit-linux-musl]
cc = "powerpc-linux-musl-gcc"
cxx = "powerpc-linux-musl-g++"
linker = "powerpc-linux-musl-gcc"

[target.powerpc64-foxkit-linux-musl]
cc = "powerpc64-linux-musl-gcc"
cxx = "powerpc64-linux-musl-g++"
linker = "powerpc64-linux-musl-gcc"

[target.x86_64-foxkit-linux-musl]
cc = "x86_64-linux-musl-gcc"
cxx = "x86_64-linux-musl-g++"
linker = "x86_64-linux-musl-gcc"
```

Reply:
> rustc will cross-compile a native compiler with your desired triple, no intermediate "unknown" triple needed.

Huh, pretty sure Rust failed me last time I tried this telling me that it didn't know that triplet yet (since Upstream Rust tarballs don't have the alpine triplets compiled in). But thanks for the pointer, will try again.

No need to mess with the config.toml, configure deals with that for us

Reply:
Don't start with a Mozilla (or Adélie) tarball. Start with your locally-installed Alpine rust, which does know the triplet.

Reply:
Can't do that because we don't exactly support crosscompiling and we don't have existing tarballs for these arches yet.

Reply:
Confirmed this again, we need the in-between step with upstream triplets.